monkey/infection_monkey/tunnel.py

```diff
@@ -191,11 +193,16 @@ def run(self):
             except socket.timeout:
                 continue
 
+        timer.set(self._calculate_timeout())
+
```

Comment:
I'm not sure the logic here is quite right. This may need to be done inside the loop, otherwise we may timeout prematurely.

Reply:
Fixed